app/jobs/data_exports/create_job.rb

```diff
@@ -0,0 +1,161 @@
+# frozen_string_literal: true
+
+module DataExports
+  # Queues the data export create job
+  class CreateJob < ApplicationJob # rubocop:disable Method/ClassLength
+    queue_as :default
+
+    def perform(data_export)
+      @manifest = ''
+      initialize_manifest(data_export.export_type)
+
+      zip = create_sample_zip(data_export) if data_export.export_type == 'sample'
+
+      attach_zip(data_export, zip)
+
+      data_export.expires_at = set_expiry
+      data_export.status = 'ready'
+      data_export.save
+    end
+
+    def initialize_manifest(export_type)
+      @manifest = if export_type == 'sample'
+                    { 'type' => 'Samples Export', 'date' => Date.current, 'children' => [] }
+                  else
+                    { 'type' => 'Analysis Export', 'date' => Date.current, 'children' => [] }
+                  end
+    end
+
+    def create_sample_zip(data_export) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
```

Comment:
This method seems to have become unnecessarily large and requires a few rubocop disables.

This method can be split up to remove all the rubocop disables by adding 2 new methods (write_attachments, and write_manifest) and moving the relevant code into the methods

- write_attachments should take params (project, sample, zip)
- write_manifest should take params (manifest, zip)

Reply:
Thanks!